homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,288 @@
+
+CONF_CLIENT_ID = "client_id"
+CONF_CLIENT_SECRET = "client_secret"
+CONF_TIME_ZONE = "time_zone"
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Required(CONF_TIME_ZONE): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Flume sensor."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    client_id = config.get(CONF_CLIENT_ID)
+    client_secret = config.get(CONF_CLIENT_SECRET)
+    time_zone = config.get(CONF_TIME_ZONE)
+    name = config.get(CONF_NAME)
+
+    FlumeDeviceList = FlumeDevice(username, password, client_id, client_secret)
+
+    try:
+        for device in FlumeDeviceList._devices:
+            if device["type"] == 2:
+                Flume = FlumeData(
+                    username,
+                    password,
+                    client_id,
+                    client_secret,
+                    device["id"],
+                    time_zone,
+                    SCAN_INTERVAL,
+                )
+                sensor_name = name + " " + device["id"]
+                add_entities([FlumeSensor(Flume, sensor_name)], True)
+    except Exception as error:
+        _LOGGER.error("Unable to setup Flume Devices: %s", error)
+        return False
+
+
+class FlumeSensor(Entity):
+    """Representation of the Flume sensor."""
+
+    def __init__(self, Flume, name):
+        """Initialize the Flume sensor."""
+        self.Flume = Flume
+        self._name = name
+        self._state = None
+        self._unit_of_measurement = "GALLONS"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the latest data and updates the states."""
+        self.Flume.update()
+        self._state = self.Flume.value
+
+
+class FlumeDevice:
+    """Get the latest data and update the states."""
+
+    def __init__(self, username, password, client_id, client_secret):
+        """Initialize the data object."""
+        self._username = username
+        self._password = password
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self._token = self.getToken()
+        self._user_id = self.getUserId()
+        self._bearer = self.getBearer()
+        self._devices = self.getDevices()
+
+    def getToken(self):
```

Comment:
This function is a duplicate of `FlumeData.getToken()`

Reply:
Thank you for the feedback! I have updated the sensor accordingly.